Purchases/RCUserIdentity.m

```diff
@@ -0,0 +1,124 @@
+//
+//  RCDeviceCache.h
+//  Purchases
+//
+// Created by RevenueCat.
+// Copyright (c) 2019 Purchases. All rights reserved.
+//
+
+#import "RCUserIdentity.h"
+#import "RCUtils.h"
+#import "RCBackend.h"
+#import "RCPurchasesErrorUtils.h"
+
+@interface RCUserIdentity ()
+
+@property (nonatomic) RCDeviceCache *deviceCache;
+
+@property (nonatomic) RCBackend *backend;
+
+@property (nonatomic) NSString *appUserID;
+
+@property (nonatomic) BOOL isAnonymous;
+
+@end
+
+@implementation RCUserIdentity
+
+- (instancetype)initWith:(RCDeviceCache *)deviceCache backend:(RCBackend *)backend
+{
+    self = [super init];
+    if (self) {
+        self.deviceCache = deviceCache;
+        self.backend = backend;
+    }
+
+    return self;
+}
+
+- (NSString *)generateRandomID
+{
+    return [NSString stringWithFormat:@"$RCAnonymousID:%@", [[NSUUID.new.UUIDString stringByReplacingOccurrencesOfString:@"-" withString:@""] lowercaseString]];
+}
+
+- (BOOL)configureAppUserID:(nullable NSString *)appUserID
+{
+    if ([self isAppUserIDAnonymous:appUserID]) {
+        RCErrorLog(@"ERROR: Cannot configure Purchases with a RevenueCat anonymous Id. See https://docs.revenuecat.com/docs/user-ids");
+        return NO;
+    }
+    BOOL isAnonymous = false;
+    if (appUserID == nil) {
+        appUserID = [self.deviceCache cachedAppUserID];
+        if (appUserID == nil) {
+            appUserID = [self generateRandomID];
+            isAnonymous = true;
+            RCDebugLog(@"Generated New App User ID - %@", appUserID);
+        } else {
+            isAnonymous = [self.deviceCache isAnonymous];
+        }
+    }
+    [self saveAppUserID:appUserID isAnonymous:isAnonymous];
+    return YES;
+}
+
+- (void)identifyAppUserID:(NSString *)appUserID withCompletionBlock:(void (^)(NSError *_Nullable error))completion
+{
+    if ([self isAppUserIDAnonymous:appUserID]) {
```

Comment:
I'm having a hard time understanding when error case will happen. Is it truly an error? If it's something that SDK should never do internally, an assertion may be more appropriate.